servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/ConnectionObserver.java

```diff
@@ -48,6 +48,22 @@
      */
     SecurityHandshakeObserver onSecurityHandshake();
```

Comment:
The motivation was to avoid having a specific protocol name, like SSL or TLS, and be consistent with other ST API: `secure()` method on the builders and `SecurityConfigurator` interfaces. `SSLSession` is a java class that we can not change, therefore decided not to stick to its name.